SystemInformer/tokprp.c

```diff
@@ -3580,6 +3588,100 @@ PPH_STRING PhFormatTokenSecurityAttributeValue(
         return string;
     }
 
+    // WIN://BGKD
+    if (ValueIndex == 0 &&
+        Attribute->ValueType == TOKEN_SECURITY_ATTRIBUTE_TYPE_INT64 &&
+        PhEqualStringRef(Name, &winBgkd, TRUE))
+    {
+        switch (Attribute->Values.Uint64[0])
+        {
+        case PsmActNotBackground:
+            PhInitFormatS(&format[0], L"Not background");
+            break;
+        case PsmActMixedHost:
+            PhInitFormatS(&format[0], L"Mixed host");
+            break;
+        case PsmActPureHost:
+            PhInitFormatS(&format[0], L"Pure host");
+            break;
+        case PsmActSystemHost:
+            PhInitFormatS(&format[0], L"System host");
+            break;
+        default:
+            PhInitFormatS(&format[0], L"Invalid");
+        }
+
+        PhInitFormatS(&format[1], L" (");
+        PhInitFormatI64D(&format[2], Attribute->Values.Uint64[0]);
+        PhInitFormatC(&format[3], L')');
+
+        return PhFormat(format, 4, 10);
+    }
+
+    // APPID://SHA1HASH, APPID://SHA256HASH, APPID://SHA256FLATHASH
+    if (ValueIndex == 0 &&
+        Attribute->ValueType == TOKEN_SECURITY_ATTRIBUTE_TYPE_OCTET_STRING &&
+        ((Attribute->Values.OctetString[0].ValueLength == 20 && PhEqualStringRef(Name, &appidSHA1, TRUE)) ||
+        (Attribute->Values.OctetString[0].ValueLength == 32 && PhEqualStringRef(Name, &appidSHA256, TRUE)) ||
+        (Attribute->Values.OctetString[0].ValueLength == 32 && PhEqualStringRef(Name, &appidSHA256Flat, TRUE))))
+    {
+        static PWCHAR hexChars = L"0123456789ABCDEF";
+        PUCHAR hashValue = Attribute->Values.OctetString[0].Value;
+        PH_STRING_BUILDER sb;
+
+        PhInitializeStringBuilder(&sb, Attribute->Values.OctetString[0].ValueLength * 2 + sizeof(UNICODE_NULL));
+
+        for (ULONG i = 0; i < Attribute->Values.OctetString[0].ValueLength; i++)
+        {
+            PhAppendCharStringBuilder(&sb, hexChars[hashValue[i] >> 4]);
+            PhAppendCharStringBuilder(&sb, hexChars[hashValue[i] & 0x0F]);
+        }
+
+        return PhFinalStringBuilderString(&sb);
```

Comment:
Please use `PhBufferToHexString`

Reply:
Ah, I was looking for something like this but didn't find and so reinvented the wheel =)